release v1.0.1 (#787)
pingcap/tidb-operator: CHANGELOG.md

- Fix `extraLabels` description in `values.yaml` ([#763](https://github.com/pingcap/tidb-operator/pull/763))
- Fix tkctl log output exception ([#797](https://github.com/pingcap/tidb-operator/pull/797))
- Add a script to remove orphaned K8s disks ([#745](https://github.com/pingcap/tidb-operator/pull/745))
- Enable cloud storage volume expansion & label local volume ([#772](https://github.com/pingcap/tidb-operator/pull/772))
- Prevent the Pump process from exiting with `0` if the Pump becomes `offline` ([#769](https://github.com/pingcap/tidb-operator/pull/769))
- Modularize GCP Terraform ([#717](https://github.com/pingcap/tidb-operator/pull/717))
- Support `binlog.pump.config` configurations for Pump and Drainer ([#693](https://github.com/pingcap/tidb-operator/pull/693))
- Remove duplicate key values ([#758](https://github.com/pingcap/tidb-operator/pull/758))
- Fix some typos ([#738](https://github.com/pingcap/tidb-operator/pull/738))
- Extend the waiting time of the `CheckManualPauseTiDB` process ([#752](https://github.com/pingcap/tidb-operator/pull/752))

> - Set the resource limit for the tidb-backup job  ([#729](https://github.com/pingcap/tidb-operator/pull/729))

Reviewer: ```suggestion
- Set the resource limit for the `tidb-backup` job  ([#729](https://github.com/pingcap/tidb-operator/pull/729))
```

- Fix e2e test compatible with v1.0.0 ([#757](https://github.com/pingcap/tidb-operator/pull/757))
- Make incremental backup test work ([#764](https://github.com/pingcap/tidb-operator/pull/764))

> - Add retry logic for LabelNodes function ([#735](https://github.com/pingcap/tidb-operator/pull/735))

Reviewer: ```suggestion
- Add retry logic for `LabelNodes` function ([#735](https://github.com/pingcap/tidb-operator/pull/735))
```

- Fix the TiKV scale-in failure in some cases ([#726](https://github.com/pingcap/tidb-operator/pull/726))
- Add affinity to Pump and Drainer ([#741](https://github.com/pingcap/tidb-operator/pull/741))

> - Refine clean logic ([#719](https://github.com/pingcap/tidb-operator/pull/719))

Reviewer: ```suggestion
- Refine cleanup logic ([#719](https://github.com/pingcap/tidb-operator/pull/719))
```

# TiDB Operator v1.0.1 Release Notes

## v1.0.1 What's New

### Action Required

> - ACTION REQUIRED: We fixed a serious bug ([#878](https://github.com/pingcap/tidb-operator/pull/878)) that could cause all `PD` and `TiKV` pods to be accidentally deleted when `kube-apiserver` fails. We strongly suggest users upgrade to `v1.0.1`.

Reviewer: ```suggestion
- ACTION REQUIRED: We fixed a serious bug ([#878](https://github.com/pingcap/tidb-operator/pull/878)) that could cause all `PD` and `TiKV` pods to be accidentally deleted when `kube-apiserver` fails. We strongly suggest that you upgrade TiDB Operator to the `v1.0.1` version.
```

> - ACTION REQUIRED: There is a bug ([#29](https://github.com/pingcap/mydumper/pull/29)) in `mydumper` which may cause the value of a column not to be included in the exported data . We strongly suggest users upgrade the backup image to `pingcap/tidb-cloud-backup:20190828`.

Reviewer 2: ```suggestion
- ACTION REQUIRED: The backup tool image [pingcap/tidb-cloud-backup](https://hub.docker.com/r/pingcap/tidb-cloud-backup) uses a forked version of [`Mydumper`](https://github.com/pingcap/mydumper). The current version `pingcap/tidb-cloud-backup:20190610` contains a serious bug that could cause the exported data lack of a column. This is fixed in [#29](https://github.com/pingcap/mydumper/pull/29). And the default image used now contains this fixed version. So users using the old version image for backup **must** upgrade to using `pingcap/tidb-cloud-backup:201908028` and do a new full backup to avoid potential data inconsistency.
```

> - ACTION REQUIRED: There is a bug ([#29](https://github.com/pingcap/mydumper/pull/29)) in `pingcap/mydumper` which may cause the value of a column not to be included in the exported data . We strongly suggest users upgrade the backup image to `pingcap/tidb-cloud-backup:20190828`.

Reviewer: ```suggestion
- ACTION REQUIRED: There is a bug ([#29](https://github.com/pingcap/mydumper/pull/29)) in `pingcap/mydumper` which may cause the value of a column not to be included in the exported data. We strongly suggest that you upgrade the backup image to `pingcap/tidb-cloud-backup:20190828`.
```

### Improvements

- Modularize gcp terraform
- Add a script to remove orphaned k8s disks
- Support `binlog.pump.config`, `binlog.drainer.config` configurations for pump and drainer
- Add `tidb-backup` job reources limits
- Add `affinity` to pump/drainer configration
- Upgrade local-volume-provisioner to `v2.3.2`
- Reduce e2e run time from `60m` to `20m`
- Prevent pump progress from an exit with 0 if the pump become `offline`
- Users can now expand cloud storage PV dynamically by increasing PVC storage size
- Add tikvGCLifeTime option to backup
- Add important parameters to `tikv.config` and `tidb.config` in `values.yaml`
- Support restore tidb cluster from a specified scheduled backup dir
- Enable cloud storage volume expansion & label local volume
- Document and improve HA algorithm
- Users can designate permitting host in chart values.tidb.permitHost
- Add zone label and reserved resources arguments to kubelet
- Update the default backup image to `pingcap/tidb-cloud-backup:20190828`

### Bug fixes

> - Fix tikv scale in failure in some cases after tikv failover

Reviewer 2: ```suggestion
- Fix tikv scale-in failure in some cases after tikv failover
```

- Fix error handling for UpdateService
- Fix some orphan pods cleaner bugs
- Fix setting StatefulSet partition bug
- Fix ad-hoc full backup failure due to incorrect claimName
- Fix pump offline: pump progress will exit with `0` if offline
- Fix tikv scale in failure in some cases
- Fix an incorrect condition judgment

## Detailed Bug Fixes and Changes

- Update tidb-cloud-backup image tag ([#846](https://github.com/pingcap/tidb-operator/pull/846))
- Add tidb permit host option ([#779](https://github.com/pingcap/tidb-operator/pull/779))
- Add zone label and reserved resources for nodes ([#871](https://github.com/pingcap/tidb-operator/pull/871))
- Fix some orphan pods cleaner bugs ([#878](https://github.com/pingcap/tidb-operator/pull/878))
- Fix setting StatefulSet partition bug ([#830](https://github.com/pingcap/tidb-operator/pull/830))
- Add tikvGCLifeTime option ([#835](https://github.com/pingcap/tidb-operator/pull/835))
- Add recommendations options to mydumper ([#828](https://github.com/pingcap/tidb-operator/pull/828))
- Fix ad-hoc full backup failure due to incorrect claimName ([#836](https://github.com/pingcap/tidb-operator/pull/836))

> - Improve tkctl get command output ([#822](https://github.com/pingcap/tidb-operator/pull/822))

Reviewer: ```suggestion
- Improve `tkctl get` command output ([#822](https://github.com/pingcap/tidb-operator/pull/822))
```

> - Add important parameters to tikv and tidb config ([#786](https://github.com/pingcap/tidb-operator/pull/786))

Reviewer: ```suggestion
- Add important parameters to TiKV and TiDB configurations ([#786](https://github.com/pingcap/tidb-operator/pull/786))
```

> - Fix v1.0.0 don't support `binlog.drainer.config` ([#775](https://github.com/pingcap/tidb-operator/pull/775))

Reviewer: ```suggestion
- Fix the issue that `binlog.drainer.config` is not supported in v1.0.0 ([#775](https://github.com/pingcap/tidb-operator/pull/775))
```

> - Support restore tidb cluster from a specified scheduled backup dir ([#804](https://github.com/pingcap/tidb-operator/pull/804))

Reviewer: ```suggestion
- Support restoring the TiDB cluster from a specified scheduled backup directory ([#804](https://github.com/pingcap/tidb-operator/pull/804))
```

- Fix `extraLabels` description in `values.yaml` ([#763](https://github.com/pingcap/tidb-operator/pull/763))
- Fix tkctl log output exception ([#797](https://github.com/pingcap/tidb-operator/pull/797))
- Add a script to remove orphaned K8s disks ([#745](https://github.com/pingcap/tidb-operator/pull/745))
- Enable cloud storage volume expansion & label local volume ([#772](https://github.com/pingcap/tidb-operator/pull/772))

> - Pump progress will exit with 0 if offline ([#769](https://github.com/pingcap/tidb-operator/pull/769))

Reviewer: ```suggestion
- Prevent the Pump process from exiting with `0` if the Pump becomes `offline` ([#769](https://github.com/pingcap/tidb-operator/pull/769))
```

> - Modularize gcp terraform ([#717](https://github.com/pingcap/tidb-operator/pull/717))

Reviewer: ```suggestion
- Modularize GCP Terraform ([#717](https://github.com/pingcap/tidb-operator/pull/717))
```

> - Support `binlog.pump.config` configurations for pump and drainer ([#693](https://github.com/pingcap/tidb-operator/pull/693))

Reviewer: ```suggestion
- Support `binlog.pump.config` configurations for Pump and Drainer ([#693](https://github.com/pingcap/tidb-operator/pull/693))
```

- Remove duplicate key values ([#758](https://github.com/pingcap/tidb-operator/pull/758))

> - Some typo fix ([#738](https://github.com/pingcap/tidb-operator/pull/738))

Reviewer: ```suggestion
- Fix some typos ([#738](https://github.com/pingcap/tidb-operator/pull/738))
```

> - Extend the wating time of CheckManualPauseTiDB process ([#752](https://github.com/pingcap/tidb-operator/pull/752))

Reviewer: ```suggestion
- Extend the waiting time of the `CheckManualPauseTiDB` process ([#752](https://github.com/pingcap/tidb-operator/pull/752))
```
typo

> - Add tidb-backup job reources limits ([#729](https://github.com/pingcap/tidb-operator/pull/729))

Reviewer: ```suggestion
- Set the resource limit for the tidb-backup job  ([#729](https://github.com/pingcap/tidb-operator/pull/729))
```

- Fix e2e test compatible with v1.0.0 ([#757](https://github.com/pingcap/tidb-operator/pull/757))
- Make incremental backup test work ([#764](https://github.com/pingcap/tidb-operator/pull/764))
- Add retry logic for LabelNodes function ([#735](https://github.com/pingcap/tidb-operator/pull/735))

> - Fix tikv scale in failure in some cases ([#726](https://github.com/pingcap/tidb-operator/pull/726))

Reviewer: ```suggestion
- Fix the TiKV scale-in failure in some cases ([#726](https://github.com/pingcap/tidb-operator/pull/726))
```

> - Add affinity to pump/drainer ([#741](https://github.com/pingcap/tidb-operator/pull/741))

Reviewer: ```suggestion
- Add affinity to Pump and Drainer ([#741](https://github.com/pingcap/tidb-operator/pull/741))
```

- Refine clean logic ([#719](https://github.com/pingcap/tidb-operator/pull/719))
- Inject a failure by pod annotation ([#716](https://github.com/pingcap/tidb-operator/pull/716))

> - Updated readme links to point to correct pingcap.com/docs URLs for English and Chinese ([#732](https://github.com/pingcap/tidb-operator/pull/732))

Reviewer: ```suggestion
- Update README links to point to correct `pingcap.com/docs` URLs for English and Chinese ([#732](https://github.com/pingcap/tidb-operator/pull/732))
```

- Document and improve HA algorithm ([#670](https://github.com/pingcap/tidb-operator/pull/670))
- Fix an incorrect condition judgment ([#718](https://github.com/pingcap/tidb-operator/pull/718))
- Upgrade local-volume-provisioner to v2.3.2 ([#696](https://github.com/pingcap/tidb-operator/pull/696))
- Reduce e2e test run time ([#713](https://github.com/pingcap/tidb-operator/pull/713))

> - Terraform gke scale out fix ([#711](https://github.com/pingcap/tidb-operator/pull/711))

Reviewer: ```suggestion
- Fix Terraform GKE scale-out issues ([#711](https://github.com/pingcap/tidb-operator/pull/711))
```
